text/0041-Recoverable-fee-history-for-FOG-compatible-gift-codes.md

```diff
@@ -0,0 +1,137 @@
+wallet users and potentially be sent to individuals who do not yet have a MobileCoin account.
+
+This is achieved in 2 stages:
+1. Sending a TxOut to a special subaddress reserved for gift codes at u64::MAX - 2 along with a *funding memo* written 
+to the change output of the transaction that records the first 4 bytes of a Blake2b512 hash of the gift code TxOut and a
+60 byte utf-8 note that can be used to describe information about the gift code. If the sending party decides to cancel 
+this gift code TxOut (say, if it is never redeemed by a receiver) it is sent back to the sending party's change address with 
+a *cancellation memo* that includes the global index of the gift code TxOut originally sent to the reserved gift code 
+subaddress.
+
+
+2. Sending a protobuf message via grpc to another party giving the information required to locate, unmask, and spend the 
+TxOut. The receiving party "redeems" the gift code by sending it to their change subaddress along with a *sender memo* 
+that allows the receiver to record up to 64 utf-8 bytes to document any desired information about the gift code.
+
+The memos which record this information are shown in the table below:
+
+| Memo type bytes | Name                                              |
+| -----------     | -----------                                       |
+| 0x0201          | Gift code funding memo                            |
+| 0x0202          | Gift code cancellation memo                       |
+| 0x0002          | Gift code sender memo                       |
+
+However, as designed, the gift code memos do not allow any room for tracking information about the fees paid to create, 
+redeem, or cancel any of the memos.
+
+This MCIP proposes to allocate 7 bytes to each memo representing an unsigned 56 bit number which can be used to track 
+the fee amount paid in each stage of the gift code flow.
+
+# Motivation
+[motivation]: #motivation
+
+Normal recoverable transaction history isn't possible for gift flow because at the point when a gift code is funded and 
+sent to the gift code subaddress, it is not yet known what account will redeem this gift code. When funding or 
+cancelling a gift code, the sender may want to be able to have a recoverable history for the fees they paid to do so.
+
+Similarly, when a gift code is "redeemed" the amount sent to the change address will differ from the amount received in
+the protobuf message by an amount equal to the fee paid to redeem it. This may be confusing to gift code recipients if 
+they are not able to account for the fee that they paid to redeem it.
+
+Thus we propose modifications to the gift code memo types in order to create recoverable fee history.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Each memo type will have allocate 7 bytes for an unsigned 56-bit number which represents the fee amount paid in each 
```

Comment:
Should mention (and maybe link) the original RTH MCIP, that also uses 7 bytes for representing the fee.